apps/explorer/src/components/top-validators-card/TopValidatorsCard.tsx

```diff
@@ -50,91 +86,139 @@ export const STATE_DEFAULT: ValidatorState = {
+        .map((av) => {
+            const rawName = av.fields.metadata.fields.name;
+            const name = textDecoder.decode(
+                new Base64DataBuffer(rawName).getData()
+            );
+            return {
+                name: name,
+                address: av.fields.metadata.fields.sui_address,
+                pubkeyBytes: av.fields.metadata.fields.pubkey_bytes,
+                stake: av.fields.stake_amount,
+                stakePercent: getStakePercent(
+                    av.fields.stake_amount,
+                    totalStake
+                ),
+                delegation_count: av.fields.delegation_count || 0,
+            };
+        })
+        .sort((a, b) => (a.name > b.name ? 1 : -1));
 }
 
-function TopValidatorsCard({ state }: { state: ValidatorState }) {
-    const validatorsData = processValidators(
-        state.validators.fields.active_validators
+export const getStakePercent = (stake: bigint, total: bigint): number =>
+    Number(BigInt(stake) * BigInt(100)) / Number(total);
+
+const validatorsTable = (validatorsData: ValidatorState, limit?: number) => {
+    const totalStake = validatorsData.validators.fields.total_validator_stake;
+    sortValidatorsByStake(validatorsData.validators.fields.active_validators);
+    const validators = processValidators(
+        validatorsData.validators.fields.active_validators,
+        totalStake
     );
 
-    // map the above data to match the table - combine stake and stake percent
-    // limit number validators to 10
-    // TODO: add sorting
-    const tableData = {
-        data: validatorsData.splice(0, 10).map((validator) => ({
-            name: validator.name,
-            address: (
-                <Longtext
-                    text={validator.address}
-                    alttext={truncate(validator.address, 14)}
-                    category="addresses"
-                    isLink
-                />
-            ),
-            pubkeyBytes: (
-                <Longtext
-                    text={validator.pubkeyBytes}
-                    alttext={truncate(validator.pubkeyBytes, 14)}
-                    category="addresses"
-                    isLink={false}
-                />
-            ),
-        })),
+    let cumulativeStakePercent = 0;
+    const validatorsItmes = limit ? validators.splice(0, limit) : validators;
+
+    return {
+        data: validatorsItmes.map((validator) => {
+            cumulativeStakePercent += validator.stakePercent;
+            return {
+                name: (
+                    <Text variant="bodySmall" color="steel-dark">
+                        {validator.name}
+                    </Text>
+                ),
+                stake: stakeColumn(validator),
+                delegation: (
+                    <Text variant="bodySmall" color="steel-darker">
+                        {validator.stake.toString()}
+                    </Text>
+                ),
+                cumulativeStake: (
```

Comment:
I don’t see this in the design? Seems like a pretty useless piece of info too.

Reply:
removed this.